### ISSUE TITLE
[2.24] : Definition for rule 'import/extensions' was not found

### ISSUE THREAD

**Issue by mahyarmirrashed** [import-js/eslint-plugin-import#2200]
For some reason, I keep running into this issue. I have the rule clearly listed in my `.eslintrc.json` but `eslint` keeps saying the rule does not exist. The expected behaviour is no error.

Details are below for debugging purposes:

#### `package.json`
```json
{
  "dependencies": {
    "consola": "^2.15.3",
    "discord.js": "^13.1.0",
    "dotenv": "^10.0.0"
  },
  "devDependencies": {
    "@types/node": "^16.7.1",
    "@typescript-eslint/eslint-plugin": "^4.29.2",
    "@typescript-eslint/parser": "^4.29.2",
    "eslint": "^7.32.0",
    "eslint-config-airbnb-base": "^14.2.1",
    "eslint-config-airbnb-typescript": "^13.0.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-import-resolver-typescript": "^2.4.0",
    "eslint-plugin-import": "^2.24.1",
    "eslint-plugin-prettier": "^3.4.1",
    "ts-node-dev": "^1.1.8",
    "typescript": "^4.3.5"
  },
  "scripts": {
    "build": "tsc --build tsconfig.json",
    "dev": "ts-node-dev --respawn --transpile-only --poll ./src/index.js",
    "lint": "eslint \"**/*.{ts,tsx}\"",
    "start": "node ./dist/index.js"
  }
}
```

#### Error:
```txt
❯ yarn lint
yarn run v1.22.10
$ eslint "**/*.{ts,tsx}"

C:\xxx\src\client\Client.ts
  1:1  error  Definition for rule 'import/extensions' was not found  import/extensions

C:\xxx\src\index.ts
  1:1  error  Definition for rule 'import/extensions' was not found  import/extensions

✖ 2 problems (2 errors, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

#### `.eslintrc.json`
```json
{
  "env": {
    "browser": true,
    "es2021": true
  },
  "extends": [
    "airbnb-typescript",
    "plugin:@typescript-eslint/recommended",
    "prettier"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 12,
    "project": "tsconfig.json",
    "sourceType": "module"
  },
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/explicit-function-return-type": [
      "error",
      {
        "allowExpressions": true
      }
    ],
    "max-len": [
      "warn",
      {
        "code": 80,
        "tabWidth": 2,
        "ignoreUrls": true,
        "ignoreStrings": true,
        "ignoreTemplateLiterals": true
      }
    ],
    "import/extensions": [
      "error",
      "ignorePackages",
      {
        "js": "never",
        "jsx": "never",
        "ts": "never",
        "tsx": "never"
      }
    ],
    "import/no-extraneous-dependencies": 0,
    "react/jsx-filename-extension": 0
  }
}
```

**Comment by ljharb** [comment]
Try adding `import` to your `plugins` list - maybe `airbnb-typescript` isn't doing it? (that's not an actual airbnb project)

**Comment by mahyarmirrashed** [comment]
As @ljharb mentioned, changing the extensions helped:

```json
"extends": [
  "airbnb-base",
  "airbnb-typescript/base",
  "plugin:@typescript-eslint/recommended",
  "prettier"
]
```

This was located on the [documentation](https://www.npmjs.com/package/eslint-config-airbnb-typescript) as well.
